base/ltcmd.dtx

```diff
@@ -1273,7 +1273,7 @@
       {
         \tl_if_empty:NF \l_@@_last_delimiters_tl
           {
-            \msg_error:nnxx { cmd } { invalid-after-optional-expandably }
+            \msg_error:nnxx { xparse } { invalid-after-optional-expandably }
```

Comment:
I think a mistake!

Reply:
why? The branch can only be taken (afais) if xparse got loaded, it is called if there was an l or u in the spec before.

Reply:
If the code is `xparse`-specific it shouldn't even be in `ltcmd`, other than the stub 'you need to load `xparse` message at the 'dead end'.

Reply:
true, you probably don't even need a stub because the calling function couldn't get called unles u and l are available. But this is the issue with xparse still in l3kernel, the coordination is a little arkward which is why I just made sure that the error message says xparse package

Reply:
This also removes this particular message from the `cmd` namespace, so redirection, if used, won't affect this one message.  I don't think this improves things.  Rather, if this is only used by `xparse` it should be moved.

The move would have to be more or less synchronised, otherwise we'd have this same definition in two places and it would error (or check before defining in `xparse`).

Reply:
> It could get moved but that wouldn't change its behavior, would it? And as you say, it would need some arkward syncing without actual gain (as far as I can see) which is why I just changed the module. I don't mind keeping it "cmd" but it is a package error and for some reason the code was left in the kernel but it gets never executed unless the package is loaded.

If it is left as `xparse` no, it would behave the same (i.e. redirection wouldn't work).  I'd dare say that `xparse` messages should remain as `cmd` because, for example, in `xparse.sty` it uses the `cmd` name: https://github.com/latex3/latex3/blob/95436ed690026255060de12ddd3cf113ad51265b/l3packages/xparse/xparse.dtx#L1081-L1099, so changing to `xparse` here feels a bit awkward.  I'd leave `cmd` here...

Reply:
changed back to cmd